cmds/core/ping/ping.go

```diff
@@ -160,32 +160,43 @@ func ping1(net6 bool, host string, i uint64, waitFor time.Duration) (string, err
 	return fmt.Sprintf("%d bytes from %v: icmp_seq=%v, time=%v", amt, host, i, latency), nil
 }
 
-func main() {
-	flag.Parse()
-
-	// options without parameters (right now just: -hV)
-	if flag.NArg() < 1 {
-		optwithoutparam()
-	}
+func ping(host string) error {
 	if *packetSize < 8 {
-		log.Fatalf("packet size too small (must be >= 8): %v", *packetSize)
+		return fmt.Errorf("packet size too small (must be >= 8): %v", *packetSize)
 	}
 
 	interval := time.Duration(*intv)
-	host := flag.Args()[0]
-
+	p := GetPing()
 	// ping needs to run forever, except if '*iter' is not zero
 	waitFor := time.Duration(*wtf) * time.Millisecond
 	var i uint64
 	for i = 1; *iter == 0 || i <= *iter; i++ {
-		msg, err := ping1(*net6, host, i, waitFor)
+		msg, err := p.ping1(*net6, host, i, waitFor)
 		if err != nil {
-			log.Fatalf("ping failed: %v", err)
+			return fmt.Errorf("ping failed: %v", err)
 		}
 		if *audible {
 			msg = "\a" + msg
 		}
 		log.Print(msg)
 		time.Sleep(time.Millisecond * interval)
 	}
+	return nil
+}
+
+func main() {
+	flag.Parse()
+	if *version {
+		fmt.Fprintf(os.Stdout, "ping utility, Uroot version\n")
+		os.Exit(0)
+	}
+	// options without parameters (right now just: -hV)
+	if flag.NArg() < 1 {
```

Comment:
suggestion: for now, just contain the version in the help. I.e., if someone does a -V, it can print out the "version" message as well. Or was the -V done to be compatible with GNU ping? GNU compatibility was an explicit non-goal of this project. But maybe we need compatibility?